File: PasswordGenerator/PHP/Web/Plain/src/Storage/DatabaseAdapter.php
<?php
interface DatabaseAdapter {
    public function connect(): void;
    public function getAll(string $table): array;
    public function getById(string $table, int|string $id): ?array;
    public function create(string $table, array $data): int|string;
    public function update(string $table, int|string $id, array $data): bool;
    public function delete(string $table, int|string $id): bool;
    public function search(string $table, string $field, string $query): array;
}

File: Inboxes/Python/Web/Plain/src/storage/database_adapter.py
from abc import ABC, abstractmethod

class DatabaseAdapter(ABC):
    @abstractmethod
    def connect(self): pass
    @abstractmethod
    def init_table(self, table, columns=None): pass
    @abstractmethod
    def get_all(self, table): pass
    @abstractmethod
    def get_by_id(self, table, id): pass
    @abstractmethod
    def create(self, table, data): pass
    @abstractmethod
    def update(self, table, id, data): pass
    @abstractmethod
    def delete(self, table, id): pass
    @abstractmethod
    def search(self, table, field, query): pass
    @abstractmethod
    def close(self): pass
